### PR TITLE
gha: enable kserve smoke test on main pushes and PRs

### PR DESCRIPTION
**Pull request by dtrifiro**
- run kserve smoke test on all pushes on main/PRs
- gha: merge kserve smoke test with docker compose smoke test workflow
- gha: use https://github.com/dtrifiro/setup-kserve action to setup kind/kserve

### CODE REVIEW

File: .github/workflows/smoke-test.yml
Comment (dtrifiro) [inline .github/workflows/smoke-test.yml:1]: This has been merged into `build-and-test.yml`

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -35,52 +35,79 @@ jobs:
...
-                          ]
-                      }
-                    ]
-                  }
-                }
-              }
-          }'
-
-      - name: Setup testing namespace
+          docker pull quay.io/opendatahub/text-generation-inference:stable
```
Comment (heyselbi) [inline .github/workflows/build-and-test.yml:82]: Hey @dtrifiro, quick question, if we are testing a PR, how would we replace this image?

Comment (heyselbi) [reply .github/workflows/build-and-test.yml:82]: Actually never mind, this is caikit-tgis, so tgis would always be stable.

**Review by heyselbi — APPROVED** [review]

**Pull request merged**